underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java

```diff
@@ -149,30 +150,42 @@ public static S3AUnderFileSystem createInstance(AlluxioURI uri) {
     }
 
     AmazonS3Client amazonS3Client = new AmazonS3Client(credentials, clientConf);
+    // Set a custom endpoint.
     if (Configuration.containsKey(PropertyKey.UNDERFS_S3_ENDPOINT)) {
       amazonS3Client.setEndpoint(Configuration.get(PropertyKey.UNDERFS_S3_ENDPOINT));
     }
+    // Disable DNS style buckets, this enables path style requests.
+    if (Configuration.getBoolean(PropertyKey.UNDERFS_S3_DISABLE_DNS_BUCKETS)) {
```

Comment:
I think he was referring to the new region field you added. Since all users of this property (s3n/s3a) default to false when it is empty, we should be able to use a default value.